### PR TITLE
Skip writing data_point.zip, but still write out.osw

### PR DESCRIPTION
**Pull request by joseph-robertson**
Introduces an optional `skip_zip_results` run option (defaults to `false`). This is somewhere between the set of default run options and the `fast` mode.

Test against:
```
jrobert1@JROBERTS-31863S MINGW64 /c/OpenStudio/resstock (restructure-v3)
$ time /c/openstudio-3.2.0/bin/openstudio.exe --include /c/OpenStudio/OpenStudio-workflow-gem/ run -w workflows/example_single_family_detached_TMY.osw
```

Scenario 1: No `run_options`
```
real    0m27.367s
user    0m0.390s
sys     0m0.546s

real    0m28.960s
user    0m0.109s
sys     0m0.656s

real    0m32.135s
user    0m0.234s
sys     0m0.516s
```

Scenario 2: `fast=true`
```
real    0m25.013s
user    0m0.172s
sys     0m0.500s

real    0m29.778s
user    0m0.250s
sys     0m0.500s

real    0m31.634s
user    0m0.312s
sys     0m0.812s
```

Scenario 3: `skip_zip_results=true`
```
real    0m25.979s
user    0m0.265s
sys     0m0.437s

real    0m29.778s
user    0m0.250s
sys     0m0.500s

real    0m27.388s
user    0m0.421s
sys     0m0.515s
```

### CODE REVIEW

File: lib/openstudio/workflow/adapters/output/local.rb [hunk 1]
```diff
@@ -136,8 +136,10 @@ def communicate_objective_function(objectives, _ = nil)
 
         # Write the results of the workflow to the filesystem
         #
-        def communicate_results(directory, results)
-          zip_results(directory)
+        def communicate_results(directory, results, skip_zip_results)
+          if !skip_zip_results
+            zip_results(directory)
```
Comment (nllong) [inline lib/openstudio/workflow/adapters/output/local.rb:141]: look good!

File: spec/files/skip_zip_results_osw/measures/IncreaseRoofRValue/measure.rb [hunk 2]
```diff
@@ -0,0 +1,299 @@
+#start the measure
+class IncreaseInsulationRValueForRoofsByPercentage < OpenStudio::Ruleset::ModelUserScript
```
Comment (nllong) [inline spec/files/skip_zip_results_osw/measures/IncreaseRoofRValue/measure.rb:2]: yikes, we have some old files in here. I'm sure we could upgrade these to be `OpenStudio::Measure`, but let's do that in a different PR. Also, the dencity measures should be deprecated.

File: lib/openstudio/workflow/run.rb [hunk 3]
```diff
@@ -90,6 +90,7 @@ def self.default_jobs
       # @option user_options [Hash] :debug Print debugging messages, overrides OSW option if set, defaults to false
       # @option user_options [Hash] :energyplus_path Specifies path to energyplus executable, defaults to empty
       # @option user_options [Hash] :fast Speeds up workflow by skipping steps not needed for running simulations, defaults to false
+      # @option user_options [Hash] :skip_zip_results Skips creating the data_point.zip file, defaults to false
```
Comment (nllong) [inline lib/openstudio/workflow/run.rb:93]:
```suggestion
      # @option user_options [Hash] :skip_zip_results Skips creating the data_point.zip file. Setting to `true` can cause issues with workflows expecting .zip files to signal completion (e.g., OpenStudio Analysis Framework), defaults to false
```

**Review by nllong — APPROVED** [review]
With and without zip on the same osw:
[without zip] =  8.76s
[with zip] = 8.81s

This was on a very small results file. If there are substantial results (gigabytes) then this can be expected to save a lot of time if an entire .zip file is not needed. Also, exchanging a non-zip file over the network will be slower--so this is really a trade-off between zip time and network speeds.

**Review by nmerket — APPROVED** [review]
Do we have a way to test this in buildstockbatch?

**Comment by joseph-robertson** [comment]
@nllong suggests that we merge this PR now, and then test the new capability using the openstudio 3.2.1 release candidate that comes out.
